### PR TITLE
Feature/md bv 247

### PR DESCRIPTION
**Pull request by dzeckelev**
refactor billing

### CODE REVIEW

File: billing/monitor.go [hunk 1]
```diff
@@ -14,28 +27,21 @@ import (
 type Monitor struct {
 	db     *reform.DB
 	logger *util.Logger
+	pc     *proc.Processor
```
Comment (ababo) [inline billing/monitor.go]: Maybe to name it `pr`?

File: billing/monitor.go [hunk 2]
```diff
@@ -53,61 +59,82 @@ func (m *Monitor) Run() error {
...
-                WHERE
-                        channels.service_status IN ('pending', 'active') AND
-                        offer.unit_type = 'seconds'
-          
-                GROUP BY channels.id, offer.setup_price, offer.unit_price, offer.max_unit
-                HAVING
-                        (
-				offer.setup_price + 
-				coalesce(sum(ses.seconds_consumed), 0) * offer.unit_price 
+		FROM channels
```
Comment (ababo) [inline billing/monitor.go]: Please use this [doc](https://www.sqlstyle.guide/) to format you SQL queries.

File: billing/monitor.go [hunk 3]
```diff
@@ -53,61 +59,82 @@ func (m *Monitor) Run() error {
...
+				ON channels.id = ses.channel
+  			LEFT JOIN offerings offer
+				ON channels.offering = offer.id
+  			LEFT JOIN accounts acc
+				ON channels.agent = acc.eth_addr
+
+		WHERE channels.service_status IN ('pending', 'active')
+			AND channels.channel_status NOT IN ('pending')
+			AND offer.unit_type = 'seconds'
+			AND acc.in_use = TRUE
```
Comment (ababo) [inline billing/monitor.go]: Can you remove `= TRUE`?

File: billing/monitor.go [hunk 4]
```diff
@@ -53,61 +59,82 @@ func (m *Monitor) Run() error {
...
+    				offer.setup_price +
+    				coalesce(sum(ses.seconds_consumed), 0) *
+				offer.unit_price
 				>= channels.total_deposit
-                        ) OR 
-			(
-				coalesce(sum(ses.seconds_consumed), 0) >= offer.max_unit
-			)`
+  			) OR
+  			(
```
Comment (ababo) [inline billing/monitor.go]: No need for round brackets.

File: billing/monitor.go [hunk 5]
```diff
@@ -53,61 +59,82 @@ func (m *Monitor) Run() error {
...
+		HAVING
 			(
-				coalesce(sum(ses.units_used), 0) >= offer.max_unit
-			)`
-
-	return m.processEachChannel(query, m.terminateService)
+    				offer.setup_price +
+    				coalesce(sum(ses.units_used), 0) *
+				offer.unit_price
+    				>= channels.total_deposit
```
Comment (ababo) [inline billing/monitor.go]: Why do you break line on each expression element? This doesn't improve readability.

File: billing/monitor_test.go [hunk 6]
```diff
@@ -4,34 +4,50 @@ package billing
```
Comment (ababo) [inline billing/monitor_test.go]: Missing `// +build !nobillingtest`.

File: billing/monitor_test.go [hunk 7]
```diff
@@ -4,34 +4,50 @@ package billing
...
 		DB          *data.DBConfig
+		Job         *job.Config
 		Log         *util.LogConfig
-		BillingTest *billingTestConfig
+		Pc          *proc.Config
 	}
-	testDB  *reform.DB
-	testMon *atomicMonitor
+
+	tDB     *reform.DB
```
Comment (ababo) [inline billing/monitor_test.go]: Why not `db`, `logger`, etc.?

File: billing/monitor_ch_bill_lags_test.go [hunk 8]
```diff
@@ -0,0 +1,81 @@
+// +build !nobillingtest
```
Comment (ababo) [inline billing/monitor_ch_bill_lags_test.go]: Why not to name it `lags_test.go`?

File: billing/monitor_ch_units_based_test.go [hunk 9]
```diff
@@ -0,0 +1,130 @@
+// +build !nobillingtest
```
Comment (ababo) [inline billing/monitor_ch_units_based_test.go]: Maybe `unit_test.go`?

File: billing/monitor_test.go (hunk 6 above)
Comment (ababo) [reply billing/monitor_test.go]: Why "thumb down"?

Comment (dzeckelev) [reply billing/monitor_test.go]: `// +build !nobillingtest` The comment was always in this file

**Pull request merged**